pkg/library/libraryapplyconfiguration/equivalence.go

```diff
@@ -84,52 +87,117 @@ func EquivalentClusterApplyResultIgnoringEvents(field string, lhs, rhs SingleClu
 	return qualifiedReasons
 }
 
-func reasonForDiff(nameOfDestination string, missingInDestination []manifestclient.SerializedRequest, allSourceRequests, allDestinationRequests []manifestclient.SerializedRequestish) []string {
+// expandedMetadata is useful for describing diffs, potentially to get pushed into manifestclient
+type expandedMetadata struct {
+	metadata               manifestclient.ActionMetadata
+	fieldManager           string
+	controllerInstanceName string
+}
+
+func expandedMetadataFor(serializedRequest *manifestclient.SerializedRequest) expandedMetadata {
+	if serializedRequest == nil {
+		return expandedMetadata{}
+	}
+	metadata := serializedRequest.GetLookupMetadata()
+	fieldManager := ""
+	controllerInstanceName := ""
+
+	isApply := serializedRequest.Action == manifestclient.ActionApply || serializedRequest.Action == manifestclient.ActionApplyStatus
+	if isApply {
+		lhsOptions := &metav1.ApplyOptions{}
+		if err := yaml.Unmarshal(serializedRequest.Options, lhsOptions); err == nil {
+			// ignore err.  if it doesn't work we get the zero value and that's ok
+			fieldManager = lhsOptions.FieldManager
+		}
+	}
+
+	bodyObj := &unstructured.Unstructured{
+		Object: map[string]interface{}{},
+	}
+	if err := yaml.Unmarshal(serializedRequest.Body, &bodyObj.Object); err == nil {
+		// ignore err.  if it doesn't work we get the zero value and that's ok
+		if bodyObj != nil && len(bodyObj.GetAnnotations()["operator.openshift.io/controller-name"]) > 0 {
+			controllerInstanceName = bodyObj.GetAnnotations()["operator.openshift.io/controller-name"]
+		}
+		if bodyObj != nil && len(bodyObj.GetAnnotations()["operator.openshift.io/controller-instance-name"]) > 0 {
```

Comment:
does this annotation exist ?

Reply:
> does this annotation exist ?

It should, we should fix the annotation, because we want controller*instances* not controllers.  We can run the same controller multiple times in the same operator.